rust/automerge/src/op_tree.rs

```diff
@@ -71,78 +190,200 @@ impl OpTreeInternal {
         self.root_node.as_ref().map_or(0, |n| n.len())
     }
 
-    pub(crate) fn keys(&self) -> Option<query::Keys<'_>> {
-        if self.root_node.is_some() {
-            Some(query::Keys::new(self))
-        } else {
-            None
+    pub(crate) fn top_ops(&self, clock: Option<Clock>) -> TopOps<'_> {
+        TopOps::new(OpTreeIter::new(self), clock)
+    }
+
+    pub(crate) fn seek_update_for_op_simple<'a>(
+        &'a self,
+        meta: &OpSetMetadata,
+        op: &'a Op,
+        mut pos: usize,
+    ) -> SeekOpFound<'a> {
+        let mut iter = self.iter();
+        let mut succ = vec![];
+        let mut next = iter.nth(pos);
```

Comment:
Can we use `Iterator::skip(pos - 1)` here followed by a normal `for` loop over `iter`?